From 23b17d5c622a8f496ebfdd011149722ecd6b8ebe Mon Sep 17 00:00:00 2001
From: Evangelos Foutras <evangelos@foutrelis.com>
Date: Sun, 1 Mar 2015 21:14:25 +0200
Subject: [PATCH] Don't quote arguments passed to terminal emulators (bug
 10731)

Commit 8cd3233 added argument quoting to several helpers to properly
pass URIs that contain spaces to web browsers like Firefox.

This broke invocations such as the following:

  $ exo-open --launch TerminalEmulator screen -S test

resulting in errors similar to:

  Failed to execute child
  Failed to execute child process "screen -S test" (No such file or
  directory)

Fix this by reverting the quoting changes made to TerminalEmulator
helpers. The Sakura terminal, surprisingly enough, seems to require
quotes around the command-line passed to it so I kept it with quotes.
---
 exo-helper/helpers/aterm.desktop.in.in                      | 2 +-
 exo-helper/helpers/debian-x-terminal-emulator.desktop.in.in | 2 +-
 exo-helper/helpers/gnome-terminal.desktop.in.in             | 2 +-
 exo-helper/helpers/nxterm.desktop.in.in                     | 2 +-
 exo-helper/helpers/qterminal.desktop.in.in                  | 2 +-
 exo-helper/helpers/urxvt.desktop.in.in                      | 2 +-
 exo-helper/helpers/xfce4-terminal.desktop.in.in             | 2 +-
 exo-helper/helpers/xterm.desktop.in.in                      | 2 +-
 8 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/exo-helper/helpers/aterm.desktop.in.in b/exo-helper/helpers/aterm.desktop.in.in
index 0eeb5ab..bbe06fc 100644
--- a/exo-helper/helpers/aterm.desktop.in.in
+++ b/exo-helper/helpers/aterm.desktop.in.in
@@ -7,4 +7,4 @@ StartupNotify=false
 X-XFCE-Binaries=aterm;
 X-XFCE-Category=TerminalEmulator
 X-XFCE-Commands=%B;
-X-XFCE-CommandsWithParameter=%B -e "%s";
+X-XFCE-CommandsWithParameter=%B -e %s;
diff --git a/exo-helper/helpers/debian-x-terminal-emulator.desktop.in.in b/exo-helper/helpers/debian-x-terminal-emulator.desktop.in.in
index a4d2cd1..b4de7ee 100644
--- a/exo-helper/helpers/debian-x-terminal-emulator.desktop.in.in
+++ b/exo-helper/helpers/debian-x-terminal-emulator.desktop.in.in
@@ -7,4 +7,4 @@ StartupNotify=false
 X-XFCE-Binaries=x-terminal-emulator;
 X-XFCE-Category=TerminalEmulator
 X-XFCE-Commands=%B;
-X-XFCE-CommandsWithParameter=%B -e "%s";
+X-XFCE-CommandsWithParameter=%B -e %s;
diff --git a/exo-helper/helpers/gnome-terminal.desktop.in.in b/exo-helper/helpers/gnome-terminal.desktop.in.in
index f6af417..bb31360 100644
--- a/exo-helper/helpers/gnome-terminal.desktop.in.in
+++ b/exo-helper/helpers/gnome-terminal.desktop.in.in
@@ -7,4 +7,4 @@ StartupNotify=true
 X-XFCE-Binaries=gnome-terminal;gnometerminal;
 X-XFCE-Category=TerminalEmulator
 X-XFCE-Commands=%B;
-X-XFCE-CommandsWithParameter=%B -x "%s";
+X-XFCE-CommandsWithParameter=%B -x %s;
diff --git a/exo-helper/helpers/nxterm.desktop.in.in b/exo-helper/helpers/nxterm.desktop.in.in
index b095f5c..9dbd157 100644
--- a/exo-helper/helpers/nxterm.desktop.in.in
+++ b/exo-helper/helpers/nxterm.desktop.in.in
@@ -7,4 +7,4 @@ StartupNotify=false
 X-XFCE-Binaries=nxterm;
 X-XFCE-Category=TerminalEmulator
 X-XFCE-Commands=%B;
-X-XFCE-CommandsWithParameter=%B -e "%s";
+X-XFCE-CommandsWithParameter=%B -e %s;
diff --git a/exo-helper/helpers/qterminal.desktop.in.in b/exo-helper/helpers/qterminal.desktop.in.in
index 05aca6b..a456bee 100644
--- a/exo-helper/helpers/qterminal.desktop.in.in
+++ b/exo-helper/helpers/qterminal.desktop.in.in
@@ -7,4 +7,4 @@ StartupNotify=false
 X-XFCE-Binaries=qterminal;
 X-XFCE-Category=TerminalEmulator
 X-XFCE-Commands=%B;
-X-XFCE-CommandsWithParameter=%B -e "%s";
+X-XFCE-CommandsWithParameter=%B -e %s;
diff --git a/exo-helper/helpers/urxvt.desktop.in.in b/exo-helper/helpers/urxvt.desktop.in.in
index b90217a..d442e42 100644
--- a/exo-helper/helpers/urxvt.desktop.in.in
+++ b/exo-helper/helpers/urxvt.desktop.in.in
@@ -7,4 +7,4 @@ StartupNotify=false
 X-XFCE-Binaries=urxvt;
 X-XFCE-Category=TerminalEmulator
 X-XFCE-Commands=%B;
-X-XFCE-CommandsWithParameter=%B -e "%s";
+X-XFCE-CommandsWithParameter=%B -e %s;
diff --git a/exo-helper/helpers/xfce4-terminal.desktop.in.in b/exo-helper/helpers/xfce4-terminal.desktop.in.in
index a9444cd..1b10ee0 100644
--- a/exo-helper/helpers/xfce4-terminal.desktop.in.in
+++ b/exo-helper/helpers/xfce4-terminal.desktop.in.in
@@ -7,4 +7,4 @@ StartupNotify=true
 X-XFCE-Binaries=xfce4-terminal;Terminal;xterminal;
 X-XFCE-Category=TerminalEmulator
 X-XFCE-Commands=%B;
-X-XFCE-CommandsWithParameter=%B -x "%s";
+X-XFCE-CommandsWithParameter=%B -x %s;
diff --git a/exo-helper/helpers/xterm.desktop.in.in b/exo-helper/helpers/xterm.desktop.in.in
index 3697935..b2bc89c 100644
--- a/exo-helper/helpers/xterm.desktop.in.in
+++ b/exo-helper/helpers/xterm.desktop.in.in
@@ -7,4 +7,4 @@ StartupNotify=false
 X-XFCE-Binaries=xterm;
 X-XFCE-Category=TerminalEmulator
 X-XFCE-Commands=%B;
-X-XFCE-CommandsWithParameter=%B -e "%s";
+X-XFCE-CommandsWithParameter=%B -e %s;
-- 
2.3.1

